ias-verify/src/utils.rs

```diff
@@ -4,14 +4,14 @@ pub fn safe_indexing(data: &[u8], start: usize, end: usize) -> Result<&[u8], &'s
         return Err("Illegal indexing");
     }
     if data.len() < end {
-        return Err("Index would be out of bounds");
+        return Err("Index out of bounds");
     }
     Ok(&data[start..end])
 }
 
 pub fn safe_indexing_one(data: &[u8], idx: usize) -> Result<u8, &'static str> {
-    if data.len() < idx {
-        return Err("Index would be out of bounds");
+    if data.len() <= idx {
+        return Err("Index out of bounds");
```

Comment:
Good catch! probably by writing unit tests? 😄 

Does this functionality not already exist somewhere in the standard library? Seems surprising to me, that we have to write code like this - is there not a `.get()` method on `slice` that returns an `Option`?

Reply:
Would love to take the credits, but this was found in the security review of polkadex.

Oh interesting, I am 99% sure that this did not exist 2 years ago. Will incorporate that. Thanks for the hint!

Reply:
You have my review, please merge at your discretion

Reply:
Haha, I forgot. Your link is part of the standard-lib, which is why we can't use it. :D